src/components/Homepage.tsx

```diff
@@ -11,13 +12,182 @@ import { PostType } from "../lib/types";
 
 const Homepage = () => {
   const navigateTo = useNavigate();
+  const [search, updateSearch] = useState("");
   const [query, updateQuery] = useState("");
+  const [selected, setSelected] = useState(false);
 
   const submitHandler: React.FormEventHandler<HTMLFormElement> = (event) => {
     event.preventDefault();
 
-    navigateTo(`/facebook?q=${query}`);
+    if (selected) navigateTo(`/facebook?q=${query}`);
+    else navigateTo(`/facebook?q=${search}`);
   };
+  const [name, updateName] = useState("");
+  const [unix, updateUnix] = useState("");
+  const [country, updateCountry] = useState("");
+  const [state, updateState] = useState("");
+  const [city, updateCity] = useState("");
+  const [button1, setButton1] = useState(false);
```

Comment:
what is button1-7? Fix naming to like professorFilterSelected, right now this makes the code very difficult to understand